common/idp.h

```diff
@@ -285,6 +285,7 @@ using service = std::tuple<balancer_service_id_t, /// service id
                            tCounterId, ///< size 4
```

Comment:
The commit message is... well... Works. Maybe. :)
I think you should squash some commits together to make reviewing easier. For example, start with a commit that refactors the old code, followed by a commit that adds chash. Make sure all corrections are included in the respective commits -- don’t leave fixes for later commits. Instead, squash those fixes so that each commit is complete and correct.